src/test/java/org/apache/maven/shared/filtering/PropertyUtilsTest.java

```diff
@@ -171,4 +176,48 @@ private void assertWarn(Logger mock, String... expected) {
             assertTrue(messages.contains(str));
         }
     }
+
+    public void testNonCircularReferences1Var3Times() throws IOException {
+        File basicProp = new File(testDirectory, "non-circular.properties");
+
+        if (basicProp.exists()) {
+            basicProp.delete();
+        }
+
+        basicProp.createNewFile();
+        try (FileWriter writer = new FileWriter(basicProp)) {
```

Comment:
try to avoid assuming the default character encoding; e.g. by using an OutputStreamWriter directly inztead